File: tui-py/resume_command.py
"""Builds the shell command to resume (or fork) a Claude Code session.

Mirrors `src/lib/resumeCommand.ts` — same POSIX single-quoting for `cwd`,
with an embedded `'` escaped as `'\\''`.
"""

from __future__ import annotations


def _shq(s: str) -> str:
    return "'" + s.replace("'", "'\\''") + "'"


def resume_command(cwd: str, session_id: str, fork: bool = False) -> str:
    base = f"cd {_shq(cwd)} && claude --resume {session_id}"
    return f"{base} --fork-session" if fork else base

File: src/lib/resumeCommand.ts
/** Shell-single-quote a string (POSIX): wrap in '…', escape embedded ' as '\''. */
function shq(s: string): string {
  return `'${s.replace(/'/g, "'\\''")}'`;
}

/** The command to paste in a terminal to resume (or fork) a session in its cwd. */
export function resumeCommand(cwd: string, sessionId: string, opts?: { fork?: boolean }): string {
  const base = `cd ${shq(cwd)} && claude --resume ${sessionId}`;
  return opts?.fork ? `${base} --fork-session` : base;
}
